make/common/modules/LauncherCommon.gmk

```diff
@@ -70,19 +66,25 @@ define SetupBuildLauncherBody
     $1_OPTIMIZATION := LOW
   endif
 
-  ifeq ($$($1_MAIN_MODULE), )
-    $1_MAIN_MODULE := $(MODULE)
-  endif
+  $1_MAIN_MODULE := $(MODULE)
 
   ifneq ($$($1_MAIN_CLASS), )
     $1_JAVA_ARGS += -Xms8m
     $1_LAUNCHER_CLASS := -m $$($1_MAIN_MODULE)/$$($1_MAIN_CLASS)
   endif
 
-  ifneq ($$($1_EXTRA_JAVA_ARGS), )
-    $1_EXTRA_JAVA_ARGS_STR := '{ $$(strip $$(foreach a, \
-      $$(addprefix -J, $$($1_EXTRA_JAVA_ARGS)), "$$a"$(COMMA) )) }'
-    $1_CFLAGS += -DEXTRA_JAVA_ARGS=$$($1_EXTRA_JAVA_ARGS_STR)
+  ifeq ($$($1_EXPAND_CLASSPATH_WILDCARDS), true)
+    $1_CFLAGS += -DEXPAND_CLASSPATH_WILDCARDS
+  endif
+
+  ifeq ($$($1_ENABLE_ARG_FILES), true)
+    $1_CFLAGS += -DENABLE_ARG_FILES
+  endif
+
+  ifeq ($(call isTargetOs, windows), true)
+    ifeq ($$($1_WINDOWS_JAVAW), true)
```

Comment:
WINDOWS\_JAVAW sounds a bit clunky. Does just JAVAW work instead? Not really that important, feel free to ignore this if you prefer the existing name

Reply:
Well, it is only affecting Windows builds. It should really be "CONSOLELESS_ON_WINDOWS" or something like that, but that was even worse. I thought "javaw" was well-known enough to be understood as "consoleless". Note that previously we set this using "CFLAGS_windows := -DJAVAW" so the Windows-only indication was there but in a slightly different way.